src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -345,6 +342,47 @@ bool RPCHandlerImpl::ConvertMessageToSO(
   return true;
 }
 
+bool RPCHandlerImpl::HandleWrongMessageType(
+    smart_objects::SmartObject& output) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  switch (output[strings::params][strings::message_type].asInt()) {
+    case application_manager::MessageType::kNotification: {
+      LOG4CXX_ERROR(logger_, "Ignore wrong HMI notification");
+      return false;
+    }
+    case application_manager::MessageType::kRequest: {
+      LOG4CXX_ERROR(logger_, "Ignore wrong HMI request");
+      output.erase(strings::msg_params);
+      output[strings::params].erase(hmi_response::message);
+      output[strings::params][hmi_response::code] =
+          hmi_apis::Common_Result::INVALID_DATA;
+      output[strings::params][strings::message_type] =
+          MessageType::kErrorResponse;
+      output[strings::params][strings::error_msg] = "";
+      return false;
+    }
+    case application_manager::MessageType::kResponse: {
+      LOG4CXX_ERROR(logger_, "Received invalid data on HMI response");
+      break;
+    }
+    case application_manager::MessageType::kUnknownType: {
```

Comment:
@ZhdanovP May be it is worth to add `default` section for this case?

Reply:
@AByzhynar  Yes, good point

Reply:
@ZhdanovP I did mean instead of this `case`